### PR TITLE
sp_BlitzFirst SET LOCK_TIMEOUT 

### PR DESCRIPTION
**Pull request by sqlslinger**
SET LOCK_TIMEOUT 1000 when APPLY sys.dm_exec_query_statistics_xml to prevent mutex issues.  Set GitHub #3210

### CODE REVIEW

File: sp_BlitzFirst.sql [hunk 1]
```diff
@@ -2293,8 +2293,9 @@ If one of them is a lead blocker, consider killing that query.'' AS HowToStopit,
 				  CROSS APPLY sys.dm_exec_query_plan(r.plan_handle) AS qp ';
 
 				IF EXISTS (SELECT * FROM sys.all_objects WHERE name = 'dm_exec_query_statistics_xml')
-					SET @StringToExecute = @StringToExecute + N' OUTER APPLY sys.dm_exec_query_statistics_xml(s.session_id) qs_live ';
-				  
+				/* GitHub #3210 */
+					SET @StringToExecute = N'
+                   SET LOCK_TIMEOUT 1000 ' + @StringToExecute + N' OUTER APPLY sys.dm_exec_query_statistics_xml(s.session_id) qs_live ';
```
Comment (BrentOzar) [inline sp_BlitzFirst.sql:2298]: Errr, that can't work. This puts SET LOCK_TIMEOUT in the middle of a query. Note what happens next in the query - it's an outer apply. That won't even compile.

You can see what I mean by trying to run:

SELECT * FROM sys.databases d
SET LOCK_TIMEOUT 1000
OUTER APPLY sys.dm_exec_query_statistics_xml(d.database_id) qs_live;

That doesn't even parse. So... with all due respect, I don't think you understand the problem you're trying to solve, so let's call this one off.

**Comment by sqlslinger** [comment]
Brent,

Are you sure?  Look again.  I added the SET LOCK_TIMEOUT 1000 *before* the
existing @StringToExecute, for exactly the reason that it already had data
in it.

So if @StringToExecute contained 'SELECT * FROM sys.databases d', the
result of the statement would be

SET LOCK_TIMEOUT 1000
SELECT * FROM sys.databases d
OUTER APPLY sys.dm_exec_query_statistics_xml(d.database_id) qs_live;

As I said, this is compiled and working.

Vince

File: sp_BlitzFirst.sql (hunk 1 above)
Comment (sqlslinger) [reply sp_BlitzFirst.sql:2298]: @BrentOzar,

Can you please reopen this issue, or comment on my response?

>Brent,
>
>Are you sure?  Look again.  I added the SET LOCK_TIMEOUT 1000 *before* the
>existing @StringToExecute, for exactly the reason that it already had data
>in it.
>
>So if @StringToExecute contained 'SELECT * FROM sys.databases d', the
>result of the statement would be
>
>SET LOCK_TIMEOUT 1000
>SELECT * FROM sys.databases d
>OUTER APPLY sys.dm_exec_query_statistics_xml(d.database_id) qs_live;
>
>As I said, this is compiled and working.
>
>Vince

**Comment by BrentOzar** [comment]
Sorry about that! I misunderstood the code. Just a gentle suggestion for next time - comments sure do help when folks are trying to understand your code. ;-)

**Review by BrentOzar — COMMENTED** [review]
Sorry about that! I misunderstood the code. Just a gentle suggestion for next time - comments sure do help when folks are trying to understand your code. ;-)

**Pull request merged**